assets/js/components/module/ModuleApp.js

```diff
@@ -68,8 +68,8 @@ function ModuleApp( { moduleSlug } ) {
 				{ moduleConnected && <DateRangeSelector /> }
 			</Header>
 			<Alert module={ moduleSlug } />
-			{ shouldRenderWidget
-				? (
+			{ shouldRenderWidget && screenWidgetContext &&
```

Comment:
Perhaps we need to display a progress bar when `shouldRenderWidget` equals `true` and `screenWidgetContext` is not loaded yet (equals to `undefined`)? Plus, we may probably need to display an error if `shouldRenderWidget` is true again, but `screenWidgetContext` is null because otherwise, users will see just a blank screen.

@aaemnnosttv @felixarntz what do you think?

Reply:
@aaemnnosttv @felixarntz I'll set this back to code review. The question from Eugene is above though if you want to check it :smile:

Reply:
I think it's a good idea to consider, but probably a little out of scope for this issue. I'm also not sure it would be possible to see this in the current module app since we preload the response that is fetched by `getModules` on page load and this will always be requested immediately on module dashboards so we would never run into a situation where the user would actually wait for any substantial amount of time here.

I think the better solution for now is to render the `WidgetContextRenderer` unconditionally if the feature is enabled and handle the `null` value within that component if necessary. We shouldn't need to worry about `ModuleApp` rendering without a screen context as this will be a requirement once the widgets API is the primary layout mechanism.

TL;DR – let's remove the progress bar for now and not use `screenWidgetContext` as a condition here if we can, and handle that in the context renderer (if needed).